lib/rspec-puppet-facts.rb

```diff
@@ -228,7 +229,14 @@ def self.with_custom_facts(os, facts)
       next if fact[:options][:confine] && !fact[:options][:confine].include?(os)
       next if fact[:options][:exclude] && fact[:options][:exclude].include?(os)
 
-      facts[name] = fact[:value].respond_to?(:call) ? fact[:value].call(os, facts) : fact[:value]
+      key = name.to_sym
```

Comment:
I'm pretty sure this is a breaking change. I think I have code that does `if facts['my_custom_fact']` and this change would make that `if facts[:my_custom_fact]`. For Puppet catalog compilation it doesn't matter, but it can break test suites.

Reply:
Yeah it sounds to me like you're right.
If introducing a breaking change, wonder does it bring back up your previous point of `RSpec.configuration.facterdb_string_keys exists. We should really do a breaking change at some point to and flip it to true.`.
My proposed change may not be the way to take this further..

Reply:
what are your thoughts for reusing this @ekohl? Just for the comparison when deep merging the facts hash, or actually storing the fact name and values as strings in the facts hash?

Reply:
Doesn't it make much more sense to address it here?
https://github.com/voxpupuli/rspec-puppet-facts/blob/8beacd8455923a7e574dd75611b4d7e5217100ce/lib/rspec-puppet-facts.rb#L213
Note how it uses `.to_s` where the rest uses `.to_sym` by default.

That at least deals with the duplicate fact keys. Merging facts is another topic.

Reply:
Actually not sure how I missed this... 
I suppose we could use `RSpec.configuration.facterdb_string_keys` as a way to determine whether the key should be converted to a string in this instance, if not - default to the format supplied by the user.

Reply:
Please still respect the `RSpec.configuration.facterdb_string_keys` setting. Your current implementation creates a problem when `RSpec.configuration.facterdb_string_keys` is `true`.

Reply:
Updated